addons/hr_expense/models/hr_expense.py

```diff
@@ -946,9 +946,34 @@ def approve_expense_sheets(self):
             if not self.env.user in current_managers and not self.user_has_groups('hr_expense.group_hr_expense_user') and self.employee_id.expense_manager_id != self.env.user:
                 raise UserError(_("You can only approve your department expenses"))
 
-        responsible_id = self.user_id.id or self.env.user.id
-        self.write({'state': 'approve', 'user_id': responsible_id})
+        responsible_id = self.user_id.id or self.env.user.id    
+        notification = {
+            'type': 'ir.actions.client',
+            'tag': 'display_notification',
+            'params': {
+                'title': "You cannot approve refused sheets.",
+                'type': 'danger',  #types: success,warning,danger,info
+                'sticky': False,  #True/False will display for few seconds if false
+            },
+        }
+        sheet_to_approve = self.filtered(lambda s: s.state in ['submit', 'draft'])
+        sheet_approved = self.filtered(lambda s: s.state in ['approve', 'done', 'post', 'cancel'])
```

Comment:
Adding the canceled expenses in the sheet_approved is one way to do it indeed.
I think it's the easier and I am ok with that. We will see the PO review.

You could rename `sheet_approved` to `sheet_reviewed` and it would be consistent with the type of record this variable can take.
As all the possible states are handled, I would have:
sheet_reviewed = self.filtered(lambda s: s.state not in ['submit', 'draft'])
That way, we know that all cases are handled.

But then your default notification message could be misleading because the message is not consistent the the rest.
I would just set title and type to None.

Reply:
I totally agree you. It's much clearer.